### PR TITLE
Fix/abs mag

### PR DESCRIPTION
**Pull request by RoyWilliams**
Please remember to follow the Lasair Pull Request checklist when issuing & reviewing this pull request:

https://lsst-uk.atlassian.net/wiki/spaces/LUSC/pages/2612264961/How+to+do+Pull+Request#STEP-3%3A-Issue-a-Pull-Request-on-Github

# CHECKLIST

- Have any tests been written to test the new code in this pull request?
  - [x] Yes. I have added the command to run the test below.
  - [ ] No. I have added the reason for the lack of testing below. 
- Have you added any new documentation for the changes in this pull-request?
  - [x] Yes. I state below where the documentation lives.
  - [ ] No. No new documentation was needed.

# NOTES

- Absolute magnitude calculation is independently checked
- Unit test for absolute magnitude
- Documentation [here](https://lasair-lsst.readthedocs.io/en/main/cookbook.html#absolute-magnitude)

# CHANGES

-

### CODE REVIEW

File: tests/unit/pipeline/filter/test_features.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        },
+    ],
+    'annotations': {
+        'sherlock': [ { } ],
+    }
+}
+
+class FeatureTest(TestCase):
+    def test0_sherlock(self):
+        alert['annotations']['sherlock'][0] = {
```
Comment (HeloiseS) [inline tests/unit/pipeline/filter/test_features.py]: maybe it's because I only know the user-side data but i don't recognise this structure 'annotations'/'sherlock'. 

@genghisken can you check this is a faithful mock?

File: tests/unit/pipeline/filter/test_features.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            'psfFlux':100000,    # mag is 18.9
+            'midpointMjdTai': 60000,
+            'band': 'g',
+        },
+    ],
+    'annotations': {
+        'sherlock': [ { } ],
+    }
+}
+
```
Comment (HeloiseS) [inline tests/unit/pipeline/filter/test_features.py:16]: worth adding a test case where flux is negative to ensure it behaves as expected (whether that is returning a null value or an abs value based on the absolute flux or raising an error you have defined)

**Review by HeloiseS — CHANGES_REQUESTED** [review]
Looks sensible and we have science values in the tests which is awesome!
Adding one test to check negative flux behaviour and double checking with somone who knows the schema of the data that the tests mock what we're dealing with in prod would be good.

File: pipeline/filter/features/sherlock.py [hunk 3]
```diff
@@ -22,32 +22,57 @@ def run(self):
...
-        # using Ken Smith code from Atlas for distance modulus
-        distances = redshiftToDistance(z)
-        distanceModulus = distances['dmod']
 
         # extinction corrected lightcurve
         (lc_ecflux, lc_time, lc_band) = getECFluxTimeBand(self.alert)
-        peakAbsMag = 99
-        peakMJD    = 99
+
+        print('flux', lc_ecflux)
```
Comment (gpfrancis) [inline pipeline/filter/features/sherlock.py]: Probably shouldn't have bare print statements

File: tests/unit/pipeline/filter/test_features.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import settings
+sys.path.append('../../../../pipeline/filter')
+sys.path.append('../../../../common/schema/' + settings.SCHEMA_VERSION)
+from objects import schema as objectSchema
+import features
+from features import sherlock
+#from features.FeatureGroup import FeatureGroup
+
+
+class FeatureTest(TestCase):
```
Comment (gpfrancis) [inline tests/unit/pipeline/filter/test_features.py:17]: `FeatureTest` is a duplicate class name. This doesn't stop the tests from running, but does mess up the reporting slightly which makes it harder to see where a failure happened. Can we rename either this one or the one in `test_feature_groups`?

**Review by gpfrancis — COMMENTED** [review]
Mostly looks good. Just these two minor points.
